ibridges/meta.py

```diff
@@ -176,26 +177,51 @@ def __setitem__(self, key: Union[str, Sequence[Union[str, None]]], other: Sequen
 
         Examples
         --------
-        >>> meta["key"] = ("key", "new_value", "new_units")
-        >>> meta["key"] = ("new_key", "old_value")
+        >>> meta["key"] = "values"
+        >>> meta["key"] = "values", "units"
+        >>> meta["key", "value"] = "units"
 
         """
         if isinstance(other, str):
-            raise TypeError(
-                "Cannot set the metadata item to a single string value. "
-                f'Use meta[{key}].key = "{other}" to change only the key '
-                "for example."
-            )
-        self[key].update(*other)
+            other = [other]
+        if isinstance(key, str):
+            key = [key]
+
+        if len(key) > 2:
+            raise ValueError("Use either one or two values within the brackets [], for example "
+                             f"meta['some_key', 'some_value'] = 'some_units', got: {key}")
+
+        all_items = self.find_all(*key)
+        if all(isinstance(o, str) for o in other):
+            if len(all_items) > 1:
+                raise ValueError(f"Cannot set item with '{key}' to single item: multiple entries"
+                                 f" exist. Use meta[{key}] = [{other}] to remove all current values"
+                                 f" with new values.")
+            other = [other]  # type: ignore
+        for subset in other:
+            if not all(isinstance(s, str) for s in subset):
+                raise ValueError(
+                    "Badly formed argument to __setitem__: should be set to either string,"
+                    " list of strings or list of list of strings.")
+            if len(subset) + len(key) > 3:
+                raise ValueError(
+                    f"Too many items to create metadata triple {subset} + {key}. Use "
+                    "meta['key'] = 'value', 'units' or meta['key', 'value'] = 'units'.")
+
+        for item in all_items:
+            item.remove()
+
+        for sub in other:
+            self.add(*key, *sub)
 
     def add(self, key: str, value: str, units: Optional[str] = ""):
         """Add metadata to an item.
 
         This will never overwrite an existing entry. If the triplet already exists
         it will throw an error instead. Note that entries are only considered the same
         if all of the key, value and units are the same. Alternatively you can use the
-        :meth:`set` method to remove all entries with the same key, before adding the
-        new entry.
+        brackets [] to remove all entries with the same key, before adding the
```

Comment:
I do not understand this help. I tried to create a new meta item but the brackets in the `add` function do not work. And I think they should not. Can we simply refer to the set functionality.

If you want to overwrite all existing metadata items with a certain key please refer to [code here].